src/tmpfiles/tmpfiles.c

```diff
@@ -655,29 +670,38 @@ static int dir_cleanup(
                         }
 
                         /* Ignore ctime, we change it when deleting */
-                        if (mtime_nsec != NSEC_INFINITY && mtime_nsec >= cutoff_nsec) {
-                                char a[FORMAT_TIMESTAMP_MAX];
-                                /* Follows spelling in stat(1). */
-                                log_debug("Directory \"%s\": modify time %s is too new.",
-                                          sub_path,
-                                          format_timestamp_style(a, sizeof(a), mtime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
-                                continue;
+                        if (FLAGS_SET(age_by_dir, AGE_BY_MTIME)) {
+                                if (mtime_nsec != NSEC_INFINITY && mtime_nsec >= cutoff_nsec) {
```

Comment:
in the interest of keeping indentation levels low, maybe merge the two if expressions:

```c
if (FLAGS_SET(…) && mtime_nsec != … && mtime_nec >= …) {
```